exchange/filecoinretrieval/filecoinretrieval.go

```diff
@@ -85,12 +99,28 @@ func (fe *FilecoinExchange) subscriber(event datatransfer.Event, channelState da
 		return
 	}
 	fe.transfersLk.RUnlock()
+	if tf.events == nil {
```

Comment:
Should `tf.events` ever be `nil`? Otherwise, `tf` can miss out on `event`s that can effect the result of that transfer.